app/src/main/java/org/mozilla/fenix/library/downloads/DownloadFragment.kt

```diff
@@ -110,4 +189,44 @@ class DownloadFragment : LibraryPageFragment<DownloadItem>(), UserInteractionHan
             )
         }
     }
+
+    private fun getDeleteHistoryItemsOperation(items: Set<DownloadItem>): (suspend () -> Unit) {
+        return {
+            CoroutineScope(IO).launch {
+                downloadStore.dispatch(DownloadFragmentAction.EnterDeletionMode)
+                context?.components?.run {
+                    for (item in items) {
+                        requireComponents.core.store.state.downloads.values.minus(requireComponents.core.store.state.downloads[item.id])
```

Comment:
We can improve the ergonomic in ac to made it easier to delete downloads by adding some extra functionality to the [download use case  class](https://github.com/mozilla-mobile/android-components/blob/master/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadsUseCases.kt). I will file an issue for us adding this functionality.

Reply:
Thank you! That would be awesome :)

Reply:
I just put this pr https://github.com/mozilla-mobile/android-components/pull/8588